quinn-proto/src/connection/mod.rs

```diff
@@ -1011,21 +1011,19 @@ impl Connection {
 
     /// Indicate what types of frames are ready to send for the given space
     fn space_can_send(&self, space_id: SpaceId, frame_space_1rtt: usize) -> SendableFrames {
-        if self.spaces[space_id].crypto.is_some()
-            || (space_id == SpaceId::Data
-                && self.zero_rtt_crypto.is_some()
-                && self.side.is_client())
+        if self.spaces[space_id].crypto.is_none()
+            && (space_id != SpaceId::Data
+                || self.zero_rtt_crypto.is_none()
+                || self.side.is_server())
         {
-            let mut can_send = self.spaces[space_id].can_send(&self.streams);
-            if space_id == SpaceId::Data {
-                can_send.other |= self.can_send_1rtt(frame_space_1rtt);
-            }
-            if !can_send.is_empty() {
```

Comment:
This condition seems to have been dropped, is that a noop? It seems unrelated to the condition inversion.

Reply:
Split into a separate commit for clarity.